### DISCUSSION TITLE
GenPollUrl should do poll until network is up

### DISCUSSION THREAD

**Discussion by StevenDoesStuffs** [qtile/qtile#3885] [Ideas]
Currently `GenPollUrl` polls before my device connects to the network, which means it displays "No Network" for a while before it polls again. I know that the polling interval is reconfigurable but it seems kinda inefficient to poll more often just because of startup, so I think having poll repeat on a smaller interval while it fails due to no network access would be nice. 

I'd be glad to PR this. 
Few implementation thoughts:

- probably want smaller interval to be configurable
- should stop polling before the next poll is started so we don't accumulate threads

**Comment by elParaguayo** [comment]
Happy to look at a PR for this!
